lib/manager/pipenv/extract.js

```diff
@@ -18,7 +19,15 @@ function extractPackageFile(content) {
   logger.debug('pipenv.extractPackageFile()');
   let pipfile;
   try {
-    pipfile = toml.parse(content);
+    const afterDiscard = content
+      .split(os.EOL)
+      .filter(
+        str =>
+          !str.match(/{\s*extras\s*=|"\*"|'\*'|=\s*{}|editable\s*=\s*true/g)
+      )
+      .join(os.EOL);
+
+    pipfile = toml.parse(afterDiscard);
```

Comment:
In general let's try to avoid this type of regex and rely on parsing instead. The parsing was not the problem I believe, it was our handling of the post-parsed content.

Reply:
Done.